src/net_processing.cpp

```diff
@@ -1470,17 +1518,16 @@ void PeerManagerImpl::SendPings()
 
 void PeerManagerImpl::RelayTransaction(const uint256& txid, const uint256& wtxid)
 {
-    m_connman.ForEachNode([&txid, &wtxid](CNode* pnode) EXCLUSIVE_LOCKS_REQUIRED(::cs_main) {
-        AssertLockHeld(::cs_main);
-
-        CNodeState* state = State(pnode->GetId());
-        if (state == nullptr) return;
-        if (state->m_wtxid_relay) {
-            pnode->PushTxInventory(wtxid);
-        } else {
-            pnode->PushTxInventory(txid);
+    LOCK(m_peer_mutex);
+    for(auto& it : m_peer_map) {
+        Peer& peer = *it.second;
+        if (!peer.m_tx_relay) return;
```

Comment:
Why is this `return` instead of `break`? Maybe I'm misunderstanding what this is doing, but I thought that we'd continue iterating through the others to relay to them, but returning seems like we're just skipping over the rest?

Reply:
Yikes. Good catch. In a previous iteration of this PR, I had a `ForEachPeer()` helper that took a lambda, and this `return` was effectively a `continue` (the lambda would return and then be called for the remaining peers). Now fixed!